internal/pkg/instrumentation/bpf/go.opentelemetry.io/otel/traceglobal/bpf/probe.bpf.c

```diff
@@ -21,9 +21,20 @@
 
 char __license[] SEC("license") = "Dual MIT/GPL";
 
+#define MAX_ATTRIBUTES 4
 #define MAX_CONCURRENT 50
 #define MAX_SPAN_NAME_LEN 64
-#define MAX_ATTRIBUTES 4
+#define MAX_STATUS_DESCRIPTION_LEN 64
+
+struct span_description_t {
+    char buf[MAX_STATUS_DESCRIPTION_LEN];
+};
+
+typedef struct otel_status {
+	u32 code;
+    u32 reserved;
```

Comment:
question: i did a lot of playing around here to get the correct struct internal offsets, which felt strange because types in go should just be wrappers for their defined types. any idea why this would be needed in my current implementation?

Reply:
I think it can be solved using `__attribute__((packed))`. This is tricky since the memory layout in the C struct should match the one defined in Go.
`bpf2go` actually generates Go equivalent structs automatically but it is not perfect.
In these cases, another good rule of thumb is to put the small fields at the end of the struct.

Reply:
yeah `__attribute__((packed))` did the trick. thanks for that. i opted to keep the struct fields ordered as code then description, but that was really just born out of preference. let me know if you'd like me to reorder them by field size.